crates/noirc_evaluator/src/ssa_refactor/ssa_builder/mod.rs

```diff
@@ -139,6 +146,8 @@ impl FunctionBuilder {
     }
 
     /// Inserts a new instruction at the end of the current block and returns its results
+    ///
+    /// TODO: Noted elsewhere this is also doing simplification, can we do this separately? tradeoffs?
```

Comment:
Doing it separately we'd either have to:

1. Remember to call it anytime before inserting an instruction
2. Let unsimplified instructions be inserted and rely on the constant folding pass to simplify each instruction.
    - This breaks passes that rely on simplification like loop unrolling which need it to check if values are known constants or not.